src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -522,6 +522,7 @@ void mp::LXDVMImageVault::poll_download_operation(const QJsonObject& json_reply,
                           .arg(json_reply["metadata"].toObject()["id"].toString()));
 
         // Instead of polling, need to use websockets to get events
+        auto last_download_progress = -2;
```

Comment:
I use `-2` because there are two different states that are being checked here. The above reply can either return a percentage for download progress (located under `download_progress -> rootfs` in the json reply, or a percentage for metadata download progress (located under `download_progress -> metadata`. The above helper function `parse_percent_as_int()` will return `-1` when downloading metadata, so a starting value of `-2` is used so that the downloading message will still be displayed in the cli.